### PR TITLE
Updating TTD options parsing to use new node_options APIs

### PR DESCRIPTION
**Pull request by mike-kaufman**
##### Checklist

- [ ] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

### CODE REVIEW

**Comment by kfarnung** [comment]
CI: https://ci.nodejs.org/job/chakracore-test-pull-request/307/

File: src/node.cc [hunk 1]
```diff
@@ -3547,7 +3538,7 @@ int Start(int argc, char** argv) {
 
   if (s_doTTRecord || s_doTTReplay) {
     TTDFlagWarning_Cond(per_process_opts->per_isolate->per_env->eval_string.length() == 0,
-                        "Eval mode not supported in record/replay.\n");
+      "Eval mode not supported in record/replay.\n");
```
Comment (kfarnung) [inline src/node.cc]: nit: undo this indentation change.

**Comment by kfarnung** [comment]
Looks like some linter failures: https://ci.nodejs.org/job/node-test-linter/21597/console

**Comment by mike-kaufman** [comment]
OK, think I fixed linter errors.

**Comment by kfarnung** [comment]
Restarted linter leg https://ci.nodejs.org/job/node-test-linter/21598/

**Comment by kfarnung** [comment]
Linter looks good now.

**Review by kfarnung — APPROVED** [review]

**Pull request merged**